tools/dep_updaters/update-openssl.sh

```diff
@@ -9,45 +9,115 @@ cleanup() {
   exit $EXIT_CODE
 }
 
-download() {
-  if [ -z "$1" ]; then
-    echo "Error: please provide an OpenSSL version to update to"
-    echo "	e.g. ./$0 download 3.0.7+quic1"
-    exit 1
-  fi
+download_v1() {
+  LATEST_V1_TAG_NAME="$("$NODE" --input-type=module <<'EOF'
+const res = await fetch('https://api.github.com/repos/quictls/openssl/releases');
+if (!res.ok) throw new Error(`FetchError: ${res.status} ${res.statusText}`, { cause: res });
+const releases = await res.json()
+const latest = releases.find(({ tag_name }) => tag_name.includes('OpenSSL_1'));
```

Comment:
This would use `name` instead of `tag_name`, and also need to look for `quic` in the `name`, otherwise the non-quic tag may come up first in the list and mess with later comparisons to the current version